analysis.ml

```diff
@@ -228,3 +229,38 @@ let dominates (instrs : segment) : pc -> pc -> bool =
   fun pc pc' ->
     let doms = dominators pc' in
     PcSet.exists ((=) pc) doms
+
+let lifetime_analysis (instrs : segment) =
+  let introduction instr = VarSet.elements (ModedVarSet.untyped (declared_vars instr)) in
+  let elimination instr = VarSet.elements (required_vars instr) in
+  scope_analysis introduction elimination instrs
```

Comment:
It would make sense to move that definition just below `liveness_analysis`: they are related and this one does not anything declared in between.